spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/mustache/MustacheServletWebConfiguration.java

```diff
@@ -26,9 +27,11 @@
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.core.Ordered;
+import org.springframework.web.servlet.ViewResolver;
 
 @Configuration(proxyBeanMethods = false)
 @ConditionalOnWebApplication(type = Type.SERVLET)
+@ConditionalOnClass(ViewResolver.class)
```

Comment:
I think it would make the intent clearer if we checked for `MustacheViewResolver`. That will check that the class and all the classes on which it depends can be loaded.

Reply:
`MustacheViewResolver` is in `spring-boot` - so wouldn't `@ConditionalOnClass(MustacheViewResolver.class)` always match?

Reply:
Looking further into it, it wouldn't. I agree and have updated the MR accordingly. Thank you!